models/product/src/product-variant/fields-config.ts

```diff
@@ -27,7 +27,20 @@ export const restFieldsConfig: TModelFieldsConfig<TProductVariantRest> = {
 export const graphqlFieldsConfig: TModelFieldsConfig<TProductVariantGraphql> = {
   fields: {
     ...commonFieldsConfig,
-    attributesRaw: [],
+    attributesRaw: fake(() => [Attribute.random()]),
     __typename: 'ProductVariant',
   },
+  postBuild: (model, context) => {
+    // @ts-expect-error The GraphQL mode does not have a `attributes` field but we need this logic for the compatibility builder
+    if (context.isCompatMode && model.attributes) {
+      // @ts-expect-error The GraphQL mode does not have a `attributes` field but we need this logic for the compatibility builder
+      model.attributesRaw = model.attributes.map((attribute) => ({
+        ...attribute,
+        __typename: 'Attribute',
```

Comment:
Should it be the type that was used in the `builders.spec.ts`?

```suggestion
        __typename: 'RawProductAttribute',
```

Reply:
You're totally right.
Thanks for spotting that, Ahmed 🙇 

I fixed it here: d447d968429b46280f45be5081140ffab19c4577

I also added tests to the presets so this kind of issue won't happen again.